lib/plugins/aws/deploy/compile/events/apiGateway/index.js

```diff
@@ -36,7 +36,8 @@ class AwsCompileApigEvents {
         _.forEach(this.serverless.service.functions, functionObj => {
           if (functionObj.events) {
             functionObj.events.forEach(event => {
-              if (event.http) noEndpoints = false;
+              // Allow events with empty http event to validate function
+              if ({}.hasOwnProperty.call(event, 'http')) noEndpoints = false;
```

Comment:
elsint will give you a [no-prototype-builtins](http://eslint.org/docs/rules/no-prototype-builtins) error